integrations/access/opsgenie/client.go

```diff
@@ -220,11 +223,29 @@ func (og Client) GetOnCall(ctx context.Context, scheduleName string) (Responders
 	}
 	defer resp.RawResponse.Body.Close()
 	if resp.IsError() {
-		return RespondersResult{}, errWrapper(resp.StatusCode())
+		return RespondersResult{}, errWrapper(resp.StatusCode(), string(resp.Body()))
 	}
 	return result, nil
 }
 
+// CheckHealth pings opsgenie.
+func (og Client) CheckHealth(ctx context.Context) error {
+	// The heartbeat pings will respond even if the heartbeat does not exist.
+	resp, err := og.client.NewRequest().
+		SetContext(ctx).
+		SetPathParams(map[string]string{"heartbeat": heartbeatName}).
+		Get("heartbeats/{heatbeat}/ping")
```

Comment:
I'm looking at Opsgenie API and it seems like you're supposed to create the heartbeat first before you can use it?

https://docs.opsgenie.com/docs/heartbeat-api#add-heartbeat-request

If so, I don't think we can use it as a health check as we would have to ask users to set up the heartbeat first which we probably don't want to require.

Is there another API we can use to check? Maybe just try and retrieve the schedules, or some other read-only call the API key we ask for allows to use? Basically, something the plugin will have to do anyway when creating an incident.

Reply:
From what i understood from "Since Opsgenie process heartbeat ping requests asynchronously, it does not check if the heartbeat exists or not before responding.
Please note that receiving a PONG response does not necessarily mean that the heartbeat exists." it should be fine to rely on this responding even if the heartbeat does not exist.

We could also just create the heartbeat when the first healthcheck is made as an alternative?